presto-main/src/main/java/com/facebook/presto/metadata/ViewDefinition.java

```diff
@@ -31,20 +32,24 @@
     private final Optional<String> schema;
     private final List<ViewColumn> columns;
     private final Optional<String> owner;
+    private final boolean runAsInvoker;
 
     @JsonCreator
     public ViewDefinition(
             @JsonProperty("originalSql") String originalSql,
             @JsonProperty("catalog") Optional<String> catalog,
             @JsonProperty("schema") Optional<String> schema,
             @JsonProperty("columns") List<ViewColumn> columns,
-            @JsonProperty("owner") Optional<String> owner)
+            @JsonProperty("owner") Optional<String> owner,
+            @JsonProperty("runAsInvoker") boolean runAsInvoker)
     {
         this.originalSql = requireNonNull(originalSql, "originalSql is null");
         this.catalog = requireNonNull(catalog, "catalog is null");
         this.schema = requireNonNull(schema, "schema is null");
         this.columns = ImmutableList.copyOf(requireNonNull(columns, "columns is null"));
         this.owner = requireNonNull(owner, "owner is null");
+        this.runAsInvoker = runAsInvoker;
+        checkArgument(!runAsInvoker || !owner.isPresent(), "owner cannot be present with runAsInvoker");
```

Comment:
Why `owner` was optional until now?
other than that, i would expect `checkArgument(runAsInvoker == !owner.isPresent()` here

Reply:
It was optional because legacy views from before owner was added will not have them. I will also add a test for that.

Reply:
That's what i suspected. Having a test would have a bonus advantage that it would self-document